utils/exceptions.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef EXCEPTIONS_HPP
+#define EXCEPTIONS_HPP
+
+#include <QException>
+#include <QDebug>
+
+namespace jbrush
+{
+// Inherits from QException and allows the client to specify a unique string message that
+// is displayed in Qt's debugging console when raised
+class GeneralException : public QException
+{
+protected:
+    QString message;
+public:
+    GeneralException() { message = "Unidentified exception ocurred"; }
+    GeneralException(QString m) { message = m; }
+    void raise() const override
+    {
+        qDebug() << message;
+        throw *this;
+    }
+    GeneralException* clone() const override { return new GeneralException(*this); }
```

Comment:
We should have a what() method that returns the message

Reply:
Point taken!  ...Although I couldn't name it `what()`.  As it turns out, `QException` inherits from `std::exception` so it already has the `what()` method, but for some reason it doesn't have a constructor that lets you set the message.  So like I tried to write `QException("error message")` but it wouldn't let me.  Weird huh?  Anyway, I named the method `errorMessage()` instead

Reply:
We should just inherit from ``std::exception`` and not from QException